R_scripts/Recursive_annot_enhanced.R

```diff
@@ -0,0 +1,190 @@
+}
+
+sj_annot <- function(sj_tab_folder        = cwd,
+                     defined_introns_file = str_glue("{cwd}/defined_introns.bed"),
+                     sj_annot_out_folder  = str_glue("{cwd}/sj_annot_out_folder")) {
+  
+  dir.create(sj_annot_out_folder, recursive = TRUE, showWarnings = TRUE)
+  
+  message("Extracting splice junction files...")
+  # Extract all splice junction files
+  splice_files <- list.files(
+    path = sj_tab_folder,
+    pattern = "*.out.tab",
+    recursive = TRUE,
+    full.names = TRUE
+  )
+  
+  message("Processing splice junction files...")
+  # Process all splice junction files
+  splice_data <- lapply(splice_files, function(file) {
+    fread(file, header = FALSE) %>%
+      filter_splicejunct_QC()
+  }) %>%
+    bind_rows() %>%
+    mutate(n = 1) %>%
+    group_by(V1, V2, V3, V4, V5, V6) %>%
+    summarise_all(sum) %>%
+    filter(n >= length(splice_files))
+  
+  # Write the filtered splice junctions to an output file
+  filterd_sj_outfile <- str_glue("{sj_annot_out_folder}/SJ.complete.out.tab")
+  write_tsv(splice_data, filterd_sj_outfile, col_names = FALSE)
+  
+  message("Running bedtools intersect...")
+  # Intersect the splice junction file with the intron file using system command
+  intersect_command <- str_glue("bedtools intersect -wa -wb -a {filterd_sj_outfile} -b {defined_introns_file} > {sj_annot_out_folder}/sj_annot_all.bed")
+  system(intersect_command)
+  
+  message("Reading intersection output...")
+  # Read the intersection output and perform further analysis
+  junction_data <- read_tsv(str_glue("{sj_annot_out_folder}/sj_annot_all.bed"), col_names = FALSE)
+  colnames(junction_data) <- c(
+    "chromosome_measured",
+    "start_position_measured",
+    "end_position_measured",
+    "strand_measured",
+    "intron_motif",
+    "annotated",
+    "unique_mapping_count",
+    "multi_mapping_count",
+    "maximum_overhang",
+    "junction_id",
+    "gtf_refrnce_chromosome",
+    "start_position_gtf_refrnce",
+    "end_position_gtf_refrnce",
+    "feature_type",
+    "gene_id",
+    "strand_gtf_refrnce"
+  )
+  
+  junction_data <- mutate(junction_data, length = end_position_gtf_refrnce - start_position_gtf_refrnce)
+  
+  junction_data <- junction_data %>%
+    mutate(strand_measured = case_when(
+      strand_measured == 1 ~ "+",
+      strand_measured == 2 ~ "-",
+      TRUE ~ as.character(strand_measured)
+    ))
+  
+  message("Processing and saving final output...")
+  # Process and save final output
+  common_start_end <- junction_data %>%
+    filter(
+      strand_measured == strand_gtf_refrnce,
+      abs(start_position_measured - start_position_gtf_refrnce) <= 1,
+      abs(end_position_measured - end_position_gtf_refrnce) <= 1
+    )
+  
+  common_start <- junction_data %>%
+    filter(
+      strand_measured == strand_gtf_refrnce,
+      abs(start_position_measured - start_position_gtf_refrnce) <= 1,
+      end_position_measured - end_position_gtf_refrnce <= -2 
```

Comment:
What is the meaning of the -2 constant here? (Would move it to variable with some descriptibve name, the same for the other constant around (1,2))